src/Microsoft.Diagnostics.Monitoring.WebApi/Metrics/MetricsStore.cs

```diff
@@ -117,12 +207,43 @@ public async Task SnapshotMetrics(Stream outputStream, CancellationToken token)
                     string metricValue,
                     string metricLabels)
         {
-            await writer.WriteAsync(metricName);
-            if (!string.IsNullOrWhiteSpace(metricLabels))
+            if (metric is GaugePayload)
```

Comment:
All of these blocks seem to be the same except for PercentilePayload. Can you generate the suffix depending on the metric type and then use the same code to write out the payload?